docs/extend/extend-apiml/api-mediation-oidc-authentication.md

```diff
@@ -26,34 +23,58 @@ The OIDC feature is currently unavailable on ACF2 systems.
-- The API ML Gateway fetches the distributed user identity from the distributed access token and maps this user identity to the user mainframe identity using SAF.
-- The API ML Gateway calls the requested mainframe service/s with mainframe user credentials (Zowe, SAF JWT, or PassTicket) which are expected by the target mainframe service.
+### Workflow description between OICD participants
+
+1. The user accesses the agent.
+2. The user agent requests the client application without valid authentication or an access token.
+3. The client redirects the user agent to the login end-point of the distributed OIDC provider.
+4. The user is asked to provide valid credentials (authentication factors).
+5. The user provides credentials.
+6. The agent sends these credentials to the OIDC provider for validation.
+7. After successful validation of all authentication factors, the OIDC provider grants the client an Access Token.
+8. The client application replies with an access token in the `set-cookie` header.
+9. The user agent can then request from API ML Gateway the needed mainframe resources presenting the access token in the request.
+10. The Gateway validates the access token in one of two ways:
+    1. By cryptographically validating the token using the public key retrieved from the authorization server's JSON Web Key Set (JWKS) endpoint, matching the token's key ID with the key IDs provided. (`components.gateway.apiml.security.oidc.validationType: JWK`).
+
+       **Notes:** 
+       * The URL to the specific authorization server's JWKS endpoint should be set using the property `components.gateway.apiml.security.oidc.jwks.uri`.
+
+       * The interval can be set using the property `components.gateway.apiml.security.oidc.jwks.refreshInternalHours`. (The default value is one hour.)
+
+    2.  By querying the `UserInfo` endpoint to verify the token's validity and retrieve user information (`components.gateway.apiml.security.oidc.validationType: endpoint`).
+
+        **Note:** The URL to the specific authorization server's `UserInfo` endpoint should be set using the property `components.gateway.apiml.security.oidc.userInfo.uri`.
+11. The Gateway caches the valid access token.
+12. The Gateway maps the distributed identity from the access token to the z/OS identity.
+
+**When user mapping exists**
+
+13. The API ML Gateway generates mainframe user credentials (Zowe JWT, SAF IDT, or PassTicket) which are expected by the target mainframe service.
+14. Calls the API with credentials.
+15. Services validates generated mainframe credentials.
+16. The requested data is returned.
+
+**When user mapping does not exist**
+
+17. The API ML Gateway calls the requested mainframe service/s with the access token in the `OIDC-token` header.
```

Comment:
If this is an alternative sequence when mapping is not provided, shouldn't the numbering here start with 13?